Gemfile

```diff
@@ -1,4 +1,7 @@
+# frozen_string_literal: true
+
 source 'https://rubygems.org'
 
-gem "rake"
-gem "rspec"
+gem 'rake'
+gem 'rspec'
```

Comment:
can I suggest doing a pessimistic major pin with `gem 'rspec', '~> 3.0'` and the same for rubocop and rake?

Reply:
I had assumed that having a checked in Gemfile.lock would mean that I'd have to manually update the dependencies anyway with bundle update, so there was limited benefit in pinning the versions in the Gemfile. Is that not the case?

Reply:
It's a slightly nuanced argument, but bundler can only install one version of each dependency. If you imagine a case in which one of your dependences depends on a specific version range of some transitive dependency, and another one depends on a sub-incompatible range, then you have to bundle update, which unpins everything. Generally speaking, it's best practice to specify desired pinning to a major in your Gemfile, and then rely on Gemfile.lock to reproduce the build for any given individual revision.

Reply:
Fair enough! I'll pin to major versions then, thanks.